cmd/clusterctl/client/repository/repository_github.go

```diff
@@ -369,3 +407,47 @@ func (g *gitHubRepository) handleGithubErr(err error, message string, args ...in
 	}
 	return errors.Wrapf(err, message, args...)
 }
+
+func getGoproxyHost() (string, error) {
+	// fallback default
+	goproxyHost := "proxy.golang.org"
+
+	// Use url from GOPROXY variable if set.
+	// xref https://github.com/golang/go/blob/master/src/cmd/go/internal/modfetch/proxy.go
+	goproxy := os.Getenv("GOPROXY")
+	for goproxy != "" {
+		var rawURL string
+		if i := strings.IndexAny(goproxy, ",|"); i >= 0 {
+			rawURL = goproxy[:i]
+			goproxy = goproxy[i+1:]
+		} else {
+			rawURL = goproxy
+			goproxy = ""
+		}
+
+		rawURL = strings.TrimSpace(rawURL)
+		if rawURL == "" {
+			continue
+		}
+		if rawURL == "off" || rawURL == "direct" {
+			return "", fmt.Errorf("expected GOPROXY environment variable to not be set to off or direct")
```

Comment:
Instead of returning an error in this case, we could return nil and fallback to github client